youtube_dlc/extractor/niconico.py

```diff
@@ -267,6 +304,14 @@ def yesno(boolean):
             'vbr': float_or_none(video_quality.get('bitrate'), 1000),
             'height': resolution.get('height'),
             'width': resolution.get('width'),
+            'quality': -2 if 'low' in format_id else -1,  # Default quality value is -1
+            'heartbeat_url': heartbeat_url,
+            'heartbeat_data': heartbeat_data,
```

Comment:
How does this affect `--write-info-json`? Can you give me an example of an output JSON?

Also test if `--load-info-json` is able to restart/continue the download

Reply:
Test failed...

I have no idea now. Let me investigate about it.

Reply:
Probably because the heartbeat URL needs to be generated again. In that case, the relevant code must be moved to the downloader. The IE is not invoked when using load-info-json

Reply:
Upon further thought, this is not necessary (though still nice to have). Even for youtube extractor, the URL needs to be regenerated if it has expired. Yet, the code for it is in the extractor and not the downloader.

So don't worry about this too much. If needed, we can fix this separately in the future. Once the issue with format quality is resolved, I will merge the PR

Reply:
I managed to fix. I tested http and hls, passed both. Resume worked fine on http.

Reply:
great. I'll check it out